shared/elements/gaia_grid/js/items/mozapp.js

```diff
@@ -184,6 +184,24 @@
       return localized || this.descriptor.short_name || this.descriptor.name;
     },
 
+    asyncName: function() {
+      var userLang = document.documentElement.lang;
+
+      var ep = this.entryPoint || undefined;
+
+      return this.app.getLocalizedValue('short_name', userLang, ep).then(
+        n => n,
```

Comment:
Can we also apply the pseudolocale logic here in case we get back the English value of `short_name`?